http/h2_stream.lua

```diff
@@ -440,14 +443,18 @@ local function validate_headers(headers, is_request, nth_header, ended_stream)
 		elseif nth_header > 2 then
 			return nil, h2_errors.PROTOCOL_ERROR:new_traceback("An HTTP request consists of maximum 2 HEADER blocks", true), ce.EILSEQ
 		end
-	else
+	elseif not expect_trailer then
```

Comment:
I think a better valid fix to this whole issue might be just changing this line to:
```suggestion
	elseif nth_header == 1 then
```

and fixing up the comment to say that not all header blocks (such as trailers) will have a status field

Reply:
Doing this will let invalid header go through if at least one interim response was received. If anything, the status check should only be removed when `ended_stream` is true, as this will limit the amount of false positive.